stack/stack.go

```diff
@@ -227,6 +227,18 @@ func LookupParent(root, dir string) (*S, bool, error) {
 	return l.lookupParentStack(dir)
 }
 
+// NewProjectMetadata creates project metadata from a given rootdir and a list of stacks.
+func NewProjectMetadata(rootdir string, stacks List) project.Metadata {
```

Comment:
This function is to specifically crete the project metadata from a stack.List type, hence why it is on the stack package instead of the project pkg.